encodings/dict/src/compress.rs

```diff
@@ -20,22 +21,66 @@ use crate::DictArray;
 /// Statically assigned code for a null value.
 pub const NULL_CODE: u64 = 0;
 
-#[derive(Debug)]
-struct Value<T>(T);
+mod private {
+    use vortex_dtype::half;
+
+    /// Value serves as a wrapper type to allow us to implement Hash and Eq on all of our
+    /// primitive types directly.
+    ///
+    /// Rust does not define Hash/Eq for any of the float types due to the presence of
+    /// NaN and +/- 0. We don't care about storing multiple NaNs or zeros in our dictionaries,
+    /// so we define simple bit-wise Hash/Eq for the Value-wrapped versions of these types.
+    #[derive(Debug)]
+    pub struct Value<T>(pub T);
+
+    macro_rules! prim_value_hash {
+        ($typ:ty) => {
+            impl core::hash::Hash for Value<$typ> {
+                fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+                    self.0.hash(state);
+                }
+            }
+
+            impl PartialEq<Self> for Value<$typ> {
+                fn eq(&self, other: &Self) -> bool {
+                    self.0.eq(&other.0)
```

Comment:
so our NativePType defines `is_eq` method for all types I wonder if we could use it here? it is defined the same way you have it here but would be nice to keep them in sync